rfcs/0001-reason-message-diff.md

```diff
@@ -0,0 +1,325 @@
+Run Image:  index.docker.io/paketobuildpacks/run@sha256:34978twgsui4789et0wh9384erhsg0e798riuhgs0er7iugshe98riugeh99rgaser89
+
+Source:   Local Source 
+
+BUILDPACK ID                            BUILDPACK VERSION
+paketo-buildpacks/bellsoft-liberica     2.13.0
+paketo-buildpacks/maven                 2.3.0
+paketo-buildpacks/executable-jar        2.1.1
+paketo-buildpacks/apache-tomcat         1.5.0
+paketo-buildpacks/dist-zip              1.4.0
+paketo-buildpacks/spring-boot           2.5.0
+```
+
+------------------------------
+
+**For Reason "CONFIG"**
+
+- Print a diff beneath the reason section
+- Print the whole config, with diffs for specific lines, with "failedBuildHistoryLimit," "successBuildHistoryLimit," "apiVersion" removed (perhaps others also?)
+
+------------------------------
+
+```
+$kp build status petclinic
+Image:      index.docker.io/username/instance@sha256:fsdgh39478th0g9tuisodfhgvns938e54iunwuiehfasf9wgiauwsfx
+Status:     SUCCESS
+Reasons:    CONFIG
+            spec:
+              tag: gcr.io/PROJECT-NAME/app
+              serviceAccount: SERVICE-ACCOUNT
+              cacheSize: "CACHE-SIZE"
+              source:
+                git:
+                  url: GIT-REPOSITORY-URL
+                  revision: GIT-REVISION
+              build:
+                env:
+                  - name: ENV-VAR-NAME
+            -       value: ENV-VAR-VALUE-1
+            +       value: ENV-VAR-VALUE-2                    
+                resources:
+                  limits:
+                    cpu: CPU-LIMIT
+                    memory: MEMORY-LIMIT
+                  requests:
+                    cpu: CPU-REQUEST
+                  memory: MEMORY-REQUEST
+
+Builder:    our-special-url.jfrog.io/instance/cluster-builder:default@sha4398ths9rdpfp9sghbne985-s9ed9hgbd9rz4sh98dxbfs
+Run Image:  index.docker.io/paketobuildpacks/run@sha256:34978twgsui4789et0wh9384erhsg0e798riuhgs0er7iugshe98riugeh99rgaser89
+
+Source:   Local Source 
+
+BUILDPACK ID                            BUILDPACK VERSION
+paketo-buildpacks/bellsoft-liberica     2.13.0
+paketo-buildpacks/maven                 2.3.0
+paketo-buildpacks/executable-jar        2.1.1
+paketo-buildpacks/apache-tomcat         1.5.0
+paketo-buildpacks/dist-zip              1.4.0
+paketo-buildpacks/spring-boot           2.5.0
+```
+
+------------------------------
+
+**For Reason "STACK"**
+
+- Print a diff beneath the reason section
+- Print the stack-name after the reason in order to give context 
+
+------------------------------
+```
+$kp build status petclinic
+Image:      index.docker.io/username/instance@sha256:fsdgh39478th0g9tuisodfhgvns938e54iunwuiehfasf9wgiauwsfx
+Status:     SUCCESS
+Reason:     STACK
+            - RunImage: index.docker.io/paketobuildpacks/run@sha256:34978twgsui4789et0wh9384erhsg0e798riuhgs0er7iugshe98riugeh99rgaser892
+            + RunImage: index.docker.io/paketobuildpacks/run@sha256:fh34578tg29wb645rwhs87e5hrw0e98jtfnws9e7w378qe49tr3g39n5gwe9u5hge549w
```

Comment:
Hello from CAPI! Will this diff be included in the Image CR's status as well? It would be really great to include this stack diff in the Image's status as well to help with the making the progress of stack updates observable. It would also be a valuable thing to add to the logs for the cf-api-controllers (which handles deploying rebased images to the cluster).

Reply:
This information will likely just be metadata placed on build when the builds are scheduled.

Reply:
> It would be really great to include this stack diff in the Image's status as well to help with the making the progress of stack updates observable

Could you explain your usecase a bit more here? Where would this information be surfaced?

Reply:
> It would also be a valuable thing to add to the logs for the cf-api-controllers (which handles deploying rebased images to the cluster).

Is this cf-api-controller's logging?